### ISSUE TITLE
Champion: numeric IntPtr

### ISSUE THREAD

**Issue by jcouv** [dotnet/csharplang#6065]
# numeric IntPtr

## Summary
[summary]: #summary

`System.IntPtr` and `System.UIntPtr` become numeric types, aliased by `nint` and `nuint`, and gain the built-in status of `nint` and `nuint`.

## Motivation
[motivation]: #motivation

As part of numerics work for .NET 7, `System.IntPtr` and `System.UIntPtr` are becoming proper numeric types.
So the distinction between native integer types and their underlying types becomes unnecessary.

## Detailed design
[design]: #detailed-design

https://github.com/dotnet/csharplang/blob/main/proposals/numeric-intptr.md (currently in [draft PR](https://github.com/dotnet/csharplang/pull/6031))

## Drawbacks
[drawbacks]: #drawbacks

## Alternatives
[alternatives]: #alternatives

## Unresolved questions
[unresolved]: #unresolved-questions

**Comment by eyalalonn** [comment]
@jcouv looks like https://github.com/dotnet/csharplang/blob/main/proposals/numeric-intptr.md is broken.

**Comment by jcouv** [comment]
Yes, the spec is still only a draft PR for now. The link will work by next week.

**Comment by walczakb** [comment]
@jcouv In section [11.10 Shift Operators](https://github.com/dotnet/csharplang/blob/main/proposals/numeric-intptr.md#1110-shift-operators), the right argument is of time `int` for operator `<<` and of type `nint` for operators `>>` and `>>>`. Is this intentional or a mistake? If intentional, why?

**Comment by jcouv** [comment]
@walczakb That was a typo. Thanks for spotting. Fixed
